addons/decimal_precision/models/decimal_precision.py

```diff
@@ -13,6 +13,27 @@ class DecimalPrecision(models.Model):
         ('name_uniq', 'unique (name)', """Only one value can be defined for each given usage!"""),
     ]
 
+    @api.onchange('digits')
+    def _onchange_digits(self):
+        all_uom = self.env['product.uom'].search([])
+        accuracy_xml_id = self.env['ir.model.data'].search([('name','=','decimal_product_uom')])
```

Comment:
```python3
accuracy_xml_id = self.env['ir.model.data'].search([('name','=','decimal_product_uom')])
if self._origin.id == accuracy_xml_id.res_id:
    new_rounding = 1.0 / 10.0**self.digits
    dangerous_uom = self.env['uom.uom'].search([('rounding', '<', new_rounding)])
    if dangerous_uom:
        errors = ["- the UOM '%s' (id=%s, precision=%s). % (uom.name, str(uom.id), str(uom.rounding)) for uom in dangerous_uom" ]
        warning = {
            'title': _('Warning!'),
            'message':
                _(
                    "You are setting a Decimal Accuracy less precise than"
                    "%s"
                    "This may cause issues in quant reservations.\n"
                    "Please set a higher number of Digits."
                    % ('\n'.join(errors)
                    )
                ,
            }
        return {'warning': warning}
```
Something like that? Only one search that does not need to be filtered and the list of all failing UoM and not only one.